components/layout/construct.rs

```diff
@@ -1205,13 +1206,13 @@ impl<'a, ConcreteThreadSafeLayoutNode: ThreadSafeLayoutNode>
                                 -> ConstructionResult {
         let flotation = FloatKind::from_property(flotation);
         let marker_fragments = match node.style(self.style_context()).get_list().list_style_image {
-            list_style_image::T::Url(ref url_value) => {
+            Either::First(ref url_value) => {
                 let image_info = box ImageFragmentInfo::new(node,
                                                             url_value.url().map(|u| u.clone()),
                                                             &self.layout_context.shared);
                 vec![Fragment::new(node, SpecificFragmentInfo::Image(image_info), self.layout_context)]
             }
-            list_style_image::T::None => {
+            Either::Second(_) => {
```

Comment:
The problem with `Either` type is that, it doesn't even give you a hint of what it holds under. We should do a workaround like this,

``` rust
Either::Second(_none) => {
```

... so that we know that it really holds a `None` type.